src/Mantasflowers.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,226 @@
+        [HttpGet("create-user")]
+        public IActionResult CreateUser(string email, string password)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.CreateUserAsync(email, password);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="uid">For testing purposes: G46RSuLIvbUjaWJyokcCMDSCeVj1</param>
+        /// <returns></returns>
+        [Authorize(Roles = "admin")]
+        [HttpGet("get-user/by-uid")]
+        public IActionResult GetUserByUid(string uid)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.GetUserByUidAsync(uid);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="email">For testing purposes: johnny@john.jo</param>
+        /// <returns></returns>
+        [Authorize(Roles = "admin")]
+        [HttpGet("get-user/by-email")]
+        public IActionResult GetUserByEmail(string email)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.GetUserByEmailAsync(email);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="email"></param>
+        /// <returns></returns>
+        [Authorize]
+        [HttpGet("update-user/email")]
+        public IActionResult UpdateUserEmail(string uid, string email)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.UpdateUserEmailAsync(uid, email);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        [Authorize]
+        [HttpGet("update-user/password")]
+        public IActionResult UpdateUserPassword(string uid, string password)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.UpdateUserPasswordAsync(uid, password);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <returns></returns>
+        [Authorize(Roles = "admin")]
+        [HttpGet("delete-user")]
+        public IActionResult DeleteUserByUid(string uid)
+        {
+            return HandleException(async () =>
+            {
+                await _fbContext.DeleteUserByUidAsync(uid);
+                return Ok(ResponseMsg.UserDeleted);
+            });
+        }
+
+        private IActionResult HandleException(Func<Task<IActionResult>> tryBlock)
```

Comment:
I know we did this separately, but I would strongly encourage to only handle very specific minimum of exceptions in controllers. Leave everything else to global error handling that we will have with my PR. For specific responses/behavior, you can throw specific exceptions and act based on the type in global error handler.